BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
+     {
+         Preconditions.checkNotNull( name, "displayName" );
+-        displayName = name;
++        // Waterfall start - we're going on an adventure!
++//        displayName = name;
++        displayName = LegacyComponentSerializer.legacySection().deserialize(name);
++        // Waterfall end
+     }
+ 
+     @Override
+@@ -395,38 +414,70 @@ public final class UserConnection implements ProxiedPlayer
+     @Override
+     public void disconnect(String reason)
+     {
+-        disconnect0( TextComponent.fromLegacyText( reason ) );
++        // Waterfall start - we're going on an adventure!
++//        disconnect0( TextComponent.fromLegacyText( reason ) );
++        disconnect0(LegacyComponentSerializer.legacySection().deserialize(reason));
++        // Waterfall end
+     }
+ 
+     @Override
+     public void disconnect(BaseComponent... reason)
+     {
+-        disconnect0( reason );
++        // Waterfall start - we're going on an adventure!
++//        disconnect0( reason );
++        disconnect0(BungeeComponentSerializer.get().deserialize(reason));
++        // Waterfall end
+     }
+ 
+     @Override
+     public void disconnect(BaseComponent reason)
+     {
+-        disconnect0( reason );
++        // Waterfall start - we're going on an adventure!
++//        disconnect0( reason );
++        disconnect0(BungeeComponentSerializer.get().deserialize(new BaseComponent[]{reason}));
++        // Waterfall end
+     }
+ 
+-    public void disconnect0(final BaseComponent... reason)
+-    {
+-        if ( !ch.isClosing() )
+-        {
+-            bungee.getLogger().log( Level.INFO, "[{0}] disconnected with: {1}", new Object[]
+-            {
+-                getName(), BaseComponent.toLegacyText( reason )
+-            } );
+-
+-            ch.close( new Kick( ComponentSerializer.toString( reason ) ) );
++    // Waterfall start - we're going on an adventure!
++    @Override
++    public void disconnect(Component reason) {
++        disconnect0(reason);
++    }
++    // Waterfall end
+ 
+-            if ( server != null )
+-            {
+-                server.setObsolete( true );
+-                server.disconnect( "Quitting" );
+-            }
++    // Waterfall start - replace disconnect0 with one that uses Adventure component - we're going on an adventure!
++//    public void disconnect0(final BaseComponent... reason)
++//    {
++//        if ( !ch.isClosing() )
++//        {
++//            bungee.getLogger().log( Level.INFO, "[{0}] disconnected with: {1}", new Object[]
++//            {
++//                getName(), BaseComponent.toLegacyText( reason )
++//            } );
++//
++//            ch.close( new Kick( ComponentSerializer.toString( reason ) ) );
++//
++//            if ( server != null )
++//            {
++//                server.setObsolete( true );
++//                server.disconnect( "Quitting" );
++//            }
++//        }
++//    }
++
++    public void disconnect0(final Component reason) {
```

Comment:
just naming this one `disconnect` should be good enough

Reply:
I named it that because it replaces the old `disconnect0`